src/item_location.cpp

```
@@ -102,14 +126,14 @@ class item_location::item_on_map : public item_location::impl
         }
 
         void remove_item() override {
-            if( what == nullptr ) {
+            if( !target() ) {
```

It silently fails?  I think the caller is going to want to know about this.